transcom/mymove
# B 18737 int

## [B-18737](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18737)

## Summary

The primary goal of this ticket is to add move-level service items to a new section in the Office MTO page, accessible by TXO and SC roles. Here, they will be displayed as read-only.

Some time was taken to research the feasibility of including dollar amount prices to this page for just these items, but after some technical discovery it was decided that this change will be made as part of another future story which will affect all service items, not just those at the move level.

A revision had to be made to account for dates being generated in a sub component dependent on the status of the service item. These dates now reflect approval/rejection dates for move-level items with corresponding statuses.

Additional PO feedback requested that the MTO Service items be included in their own section away from shipments, showing once per MTO page. This has been completed and added to this PR.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the customer app. Some existing test data will not have the move-level service items created, so we will create our own.
2. Create and submit a move.
3. Log in as a service counselor, approve the move.
4. Log in as a TOO, approve the move. Then, view the MTO Page for the move, and scroll down. You should see the Move Management fee approved in it's own section. See the below screenshot for an example.
5. A revision had to be made to account for dates being generated in a sub component dependent on the status of the service item. These dates now reflect approval/rejection dates for move-level items with corresponding statuses. Please ensure they appear correctly.
6. An additional revision had to be made to show the service items separately, outside of shipments. Please ensure it appears like the updated image below.
![image](https://github.com/transcom/mymove/assets/148906585/f869fa55-90a9-4c55-afba-2fa34aafa414)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![image](https://github.com/transcom/mymove/assets/148906585/f869fa55-90a9-4c55-afba-2fa34aafa414)
